addons/im_livechat/controllers/main.py

```diff
@@ -49,13 +49,13 @@ def support_page(self, channel_id, **kwargs):
     def loader(self, channel_id, **kwargs):
         username = kwargs.get("username", _("Visitor"))
         channel = request.env['im_livechat.channel'].sudo().browse(channel_id)
-        info = request.env['im_livechat.channel'].get_livechat_info(channel.id, username=username)
+        info = channel.get_livechat_info(username=username)
         return request.render('im_livechat.loader', {'info': info, 'web_session_required': True}, headers=[('Content-Type', 'application/javascript')])
 
     @http.route('/im_livechat/init', type='json', auth="public", cors="*")
     def livechat_init(self, channel_id):
         LivechatChannel = request.env['im_livechat.channel']
-        available = len(LivechatChannel.browse(channel_id).get_available_users())
+        available = len(LivechatChannel.sudo().browse(channel_id)._get_available_users())
```

Comment:
Actually could be implified by removing LivechatChannel variable that seems unnecessary, just directly request.env[].sudo ...